### ISSUE TITLE
Enable native Redis cluster

### ISSUE THREAD

**Issue by jaketlarson** [getsentry/sentry#12067]
## Summary

There seems to be work done already on migrating to native Redis clusters. I set up a cluster locally to test but could not compile because it seems that it's manually disabled until we rename `redis_clusters` to `clusters` in utils/redis.py:
https://github.com/getsentry/sentry/commit/f41ecd3300a47f78d972f70b973f1aad0586c4cf#diff-eca64da2111d6c6353c9e16283daac43R125

## Motivation

I'm not sure what missing pieces are left before migrating to redis clusters, but we are looking for some form of persistence via the master/replica paradigm, and from what I've gathered, the Redis Blaster setup Sentry currently uses does not support that. If a node in the Redis Blaster cluster goes down, it seems like we'd lose the data entirely with connection errors thrown.

**Comment by semistone222** [comment]
I also want to use redis cluster (master/replica).

**Comment by BYK** [comment]
@wedamija @mattrobenolt @joshuarli do you know we already have this now?

**Comment by joshuarli** [comment]
Transitioning the remaining redis-blaster-based clusters to redis-cluster-based is desirable, but I don't see that happening in the short term. At least, I recently briefly attempted it for TSDB, but struggled with some rb -> rc API conversions and moved on to other things.

**Comment by mattrobenolt** [comment]
Yeah, this is very nontrivial to do and haven't had much motivation to do the work yet. We will eventually. But it hasn't been important enough for us yet.

**Comment by ilya-jetruby** [comment]
Still actual

**Comment by Nikita-Ma** [comment]
+ Still actual
